Document bounds-checking behavior in python deviations page (#5981)
numba/numba: docs/source/reference/pysemantics.rst

Deviations from Python Semantics
================================

Bounds Checking
---------------

By default, instead of causing an :class:`IndexError`, accessing an

> out-of-bound index of an array in a Numba-compiled function will return
> garbage results or cause a segfault. Bounds checking can be enabled on a

Reviewer: ```suggestion
out-of-bound index of an array in a Numba-compiled function will return
invalid values or lead to an access violation error (it's reading from invalid memory locations). Bounds checking can be enabled on a
```
(with line wrap)

specific function via the :ref:`boundscheck <jit-decorator-boundscheck>`

> option of the jit decorator. Additionally, the `NUMBA_BOUNDSCHECK` can be set
> to 0 or 1 to globally override this flag.

Reviewer: ```suggestion
option of the jit decorator. Additionally, the :envvar:`NUMBA_BOUNDSCHECK` can be set
to 0 or 1 to globally override this flag.
```

> Bounds checking will slow down typical functions so it is recommended to only
> use this flag for debugging.

Reviewer: ```suggestion
.. note:: Bounds checking will slow down typical functions so it is recommended to only
          use this flag for debugging purposes.
```
